Implement Dashboard API and Refactor Login Flow

Felaeia opened the pull request:
Refractor:
+ Changed route in login to /admin/dashboard
+ Transfered Login hadnling from loginView to auth.ts

Feat:
+ Add API Connection of Dashboard.vue for total teachers for amount of data of /teachers
+ Add API connection of Dashboard.vue total rooms for amound of data of /labratories/all

Fix:
+ issue on where their is no accessToken passed to the route after login

Chore:
+ Issue in Notification when during Login, their would be tons of notification spamming. Current temp fix is commenting notification.

Felaeia committed:
Refractor:
+ Changed route in login to /admin/dashboard
+ Transfered Login hadnling from loginView to auth.ts
+ Add API Connection of Dashboard.vue for total teachers for amount of data of /teachers
+ Add API connection of Dashboard.vue total rooms for amound of data of /labratories/all

fix:
+ issue on where their is no accessToken passed to the route after login

Chore:
+ Issue in Notification when during Login, their would be tons of notification spamming. Current temp fix is commenting notification.

Felaeia committed:
Merge branch 'develop' of https://github.com/CS31A/Lab-System-Frontend into develop

Felaeia committed:
Delete src/components/admin/Instructions.txt

Felaeia committed:
Remove XSRF token settings from axios config

Removed XSRF token cookie and header names from axios configuration.

Felaeia committed:
Delete src/stores/routes.txt

Neil-urk12 commented on src/stores/auth.ts:
Redundant headers configuration in login action

Neil-urk12 commented on src/stores/auth.ts:
redundant headers configuration in both login and logout methods

You have already 'ensured' the inclusion of credentials in the instance level of axios.
No need to add "withCredentials: true" in every request now.

Neil-urk12 commented on src/stores/auth.ts:
The headers in the login action are redundant and the other one is incorrect.
Access-Control-Allow-Credentials is a response header that the server sends to the browser, not a request header
The headers are redundant and unnecessary. Kindly remove it.

Neil-urk12 replied:
const response = await api.post('/auth/login', { username, password })
This should be the structure of the login request

Neil-urk12 commented on src/stores/auth.ts:
Don't do this bruh
```
      // Log the login response details
      const setCookieHeader = response.headers['set-cookie'] || []
      console.warn('Login Response:', {
        status: response.status,
        headers: response.headers,
        setCookie: setCookieHeader,
        cookies: document.cookie,
        data: response.data,
      })

      // If we got a successful response, set authenticated state
      if (response.data?.message === 'Login successful') {
        isAuthenticated.value = true
        // The user data might be in response.data.data
        if (response.data.data?.user) {
          user.value = response.data.data.user
        }
      }
      else {
        throw new Error('Login response missing success message')
      }
```

Cookies are not supposed to be logged or accessed at all

Neil-urk12 replied:
You can set isAuthenticated by checking if the login response body is null or not

Felaeia replied:
This was for testing the API if i was able to recieve the cookies properly lol. I logged the cookies in the console.

Neil-urk12 replied:
> This was for testing the API if i was able to recieve the cookies properly lol. I logged the cookies in the console.

You can check it via network panel in browser devtools when performing the request. Avoid doing this.

Neil-urk12 commented on src/views/LoginView.vue:
Redirecting should be dynamic
If the user's role is admin then they should be redirected to the admin dashboard.
If they're a teacher, then they should be rdirected to teacher dashboard.
So on

If they're not an admin, a teacher, or a technical staff, then they shouldn't be redirected to any private routes.

Neil-urk12 commented on src/components/admin/Dashboard.vue:
Should create a getter computed property in authStore for getting the value of isAuthenticated.
Instead of doing authStore.isAuthenticated, you should create a computed property in authStore like
const getIsAuthenticated = computed(() => isAuthenticated)

Then do authStore.getIsAuthenticated instead, so the state is reactive to any data changes.

Neil-urk12 commented on src/components/admin/Dashboard.vue:
Use Promise.allSettled instead of Promise.all

    await Promise.all([
      dashboardStore.fetchLaboratories(),
      dashboardStore.fetchTeachers(),
      // dashboardStore.fetchDashboardData()
    ])

Neil-urk12 commented on src/components/admin/Dashboard.vue:
When you access states from pinia store, access it via getter/computed property instead of directly accessing them.

Ex:
dashboardStore.stats.totalRooms,

You should have a computed property in dashboardStore for getting the totalRooms

Neil-urk12 commented on src/interfaces/interfaces.ts:
Why does laboratory interface have email attribute

Neil-urk12 commented:
Kindly include images in pull requests always, including the results of the test cases you made.
Also, avoid including file changes that are unrelated to the main changes you are making (This PR should've been about implementing dashboard api only and refactor login flow should've been another pr from another branch.

Felaeia committed:
Refractor:
+ Add logic in loginview to route depending on the data of the response of /auth/login from the api
+ Removed uneccesary headers & logs that was used
+ Transfered withCredentials:true above the headers to allow the api to accept the credentials first before the headers as it is the fix for the mentioned fix below

Fix:
+ fix issue on API not being able to accept cookies

Felaeia committed:
Merge branch 'develop' of https://github.com/Felaeia/Lab-System-Frontend into develop

Felaeia committed:
Merge pull request #2 from CS31A/develop

Merging Latest PRs to Felaeia Fork

Neil-urk12 commented:
This change is redundant and must be reverted.

<img width="594" height="379" alt="image" src="https://github.com/user-attachments/assets/94fd85d7-c0b0-4ebc-aed5-d67a8a04e0c2" />

Neil-urk12 commented:
pull request update?
